File: lc_soln/0739_daily_temperatures/solution.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {
        std::vector<int> result(temperatures.size(), 0);
        std::stack<int> stack;
        for (int i = 0; i < temperatures.size(); ++i) {
            while (!stack.empty() && temperatures[i] > temperatures[stack.top()]) {
                int j = stack.top();
                result[j] = i - j;
                stack.pop();
            }
            stack.push(i);
        }
        return result;
    }
};

int main() {
    return 0;
}

File: lc_soln/0739_daily_temperatures/solution.rs
struct Solution;

impl Solution {
    pub fn daily_temperatures(temperatures: Vec<i32>) -> Vec<i32> {
        let mut result: Vec<i32> = vec![0; temperatures.len()];
        let mut stack = vec![];
        for i in 0..temperatures.len() {
            while let Some(&j) = stack.last() && temperatures[i] > temperatures[j] {
                result[j] = (i - j) as i32;
                stack.pop();
            }
            stack.push(i);
        }
        result
    }
}

fn main() {
}
